desc/objectives/_neoclassical.py

```diff
@@ -0,0 +1,251 @@
+    weight : {float, ndarray}, optional
+        Weighting to apply to the Objective, relative to other Objectives.
+        Must be broadcastable to Objective.dim_f
+    normalize : bool, optional
+        Whether to compute the error in physical units or non-dimensionalize.
+    normalize_target : bool, optional
+        Whether target and bounds should be normalized before comparing to computed
+        values. If `normalize` is ``True`` and the target is in physical units,
+        this should also be set to True.
+    loss_function : {None, 'mean', 'min', 'max'}, optional
+        Loss function to apply to the objective values once computed. This loss function
+        is called on the raw compute value, before any shifting, scaling, or
+        normalization.
+    deriv_mode : {"auto", "fwd", "rev"}
+        Specify how to compute Jacobian matrix, either forward mode or reverse mode AD.
+        "auto" selects forward or reverse mode based on the size of the input and output
+        of the objective. Has no effect on self.grad or self.hess which always use
+        reverse mode and forward over reverse mode respectively.
+    grid : Grid, optional
+        Collocation grid to evaluate flux surface averages.
+        Should have poloidal and toroidal resolution.
+    alpha : ndarray
+        Unique coordinate values for field line poloidal angle label alpha.
+    zeta : ndarray
+        Unique coordinate values for field line following coordinate zeta. Must be
+        strictly increasing. A good reference density is 100 knots per toroidal transit.
+        For axisymmetric devices only one toroidal transit is necessary. Otherwise,
+        more toroidal transits will give more accurate result, with diminishing returns.
+    num_quad : int
+        Resolution for quadrature of bounce integrals. Default is 32.
+    num_pitch : int
+        Resolution for quadrature over velocity coordinate, preferably odd.
+        Default is 99. Profile will look smoother at high values.
+        (If computed on many flux surfaces and small oscillations is seen
+        between neighboring surfaces, increasing this will smooth the profile).
+    batch : bool
+        Whether to vectorize part of the computation. Default is true.
+    num_well : int
+        Maximum number of wells to detect for each pitch and field line.
+        Default is to detect all wells, but due to limitations in JAX this option
+        may consume more memory. Specifying a number that tightly upper bounds
+        the number of wells will increase performance.
+    name : str, optional
+        Name of the objective function.
+
+    """
+
+    _coordinates = "r"
+    _units = "~"
+    _print_value_fmt = "Effective ripple ε¹ᐧ⁵: "
+
+    def __init__(
+        self,
+        eq,
+        target=0,
+        bounds=None,
+        weight=1,
+        normalize=True,
+        normalize_target=True,
+        loss_function=None,
+        deriv_mode="auto",
+        grid=None,
```

Comment:
what is the grid itself used for? Do we just want to pass in rho values directly?

Reply:
Right now it's just the grid used to compute $\iota$ and other flux quantities. I will change it to be the grid that everything is computed on, then interpolate from that grid to the field lines

Reply:
so that we don't have to build new transforms inside objective.compute

Reply:
Right now, the API of the objective will not change once the implementation switches to using #1119 . If we remove the grid, then the API will change once 1119 gets merged.